tests/shared/src/test/scala/org/http4s/PathSuite.scala

```diff
@@ -19,8 +19,21 @@ package org.http4s
 import cats.kernel.laws.discipline._
 import org.http4s.Uri.Path
 import org.http4s.laws.discipline.arbitrary._
+import org.scalacheck.Prop._
 
 class PathSuite extends Http4sSuite {
   checkAll("Order[Path]", OrderTests[Path].order)
   checkAll("Semigroup[Path]", SemigroupTests[Path].semigroup)
+
+  test("equals should be consistent with equals") {
+    forAll { (a: Path, b: Path) =>
+      (a == b) == (a.segments == b.segments) == (a.absolute == b.absolute) == (a.endsWithSlash == b.endsWithSlash)
```

Comment:
Should those be `&&` ?

```suggestion
      (a == b) == ((a.segments == b.segments) && (a.absolute == b.absolute) && (a.endsWithSlash == b.endsWithSlash))
```

Reply:
And maybe an implies actually?

```suggestion
      (a == b) ==> (a.segments == b.segments) && (a.absolute == b.absolute) && (a.endsWithSlash == b.endsWithSlash)
```

Reply:
```
  /** Returns a new property that holds if and only if both this
   *  and the given property generates a result with the exact
   *  same status. Note that this means that if one of the properties is
   *  proved, and the other one passed, then the resulting property
   *  will fail. */
  def ==(p: => Prop) = this.flatMap { r1 =>
    p.map { r2 =>
      mergeRes(r1, r2, if(r1.status == r2.status) True else False)
    }
  }
```
:facepalm: 
Ignore me - thanks for pointing that out!

Reply:
Unfortunately it still fails :-/